src/node/internal/internal_http_agent.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+import { EventEmitter } from 'node-internal:events';
+import { validateOneOf, validateNumber } from 'node-internal:validators';
+import { ERR_METHOD_NOT_IMPLEMENTED } from 'node-internal:internal_errors';
+import type {
+  RequestOptions,
+  Agent as _Agent,
+  AgentOptions,
+  ClientRequest,
+  IncomingMessage,
+} from 'node:http';
+import type { Socket } from 'node:net';
```

Comment:
This could probably use some explanatory comments explaining that we aren't really intending to support the `Agent` API right now beyond providing a very limited stub impl